Content.Server/_NF/GameRule/NfAdventureRuleSystem.cs

```diff
@@ -110,231 +123,227 @@ private void OnPlayerSpawningEvent(PlayerSpawnCompleteEvent ev)
+        foreach (var proto in optionalList)
         {
-            var meta = EnsureComp<MetaDataComponent>(depotUid2s[0]);
-            _meta.SetEntityName(depotUid2s[0], "Tinnia's Rest", meta);
-            _shuttle.SetIFFColor(depotUid2s[0], factionColor);
+            allLocationList.Remove(proto);
         }
+        GenerateOptionals(optionalList);
 
-        if (_map.TryLoad(mapId, church, out var churchUids, new MapLoadOptions
-            {
-                Offset = -tinniaOffset
-            }))
-        {
-            var meta = EnsureComp<MetaDataComponent>(churchUids[0]);
-            _meta.SetEntityName(churchUids[0], "Omnichurch Beacon", meta);
-            _shuttle.SetIFFColor(churchUids[0], factionColor);
-        }
+        // the remainder are done on a per-poi-per-group basis
+        var uniqueList = allLocationList;
 
-        if (_map.TryLoad(mapId, lpbravo, out var lpbravoUids, new MapLoadOptions
-            {
-                Offset = _random.NextVector2(2150f, 4850f)
-            }))
-        {
-            if (_prototypeManager.TryIndex<GameMapPrototype>("LPBravo", out var stationProto))
-            {
-                _station.InitializeNewStation(stationProto.Stations["LPBravo"], lpbravoUids);
-            }
+        GenerateUniques(uniqueList);
 
-            var meta = EnsureComp<MetaDataComponent>(lpbravoUids[0]);
-            _meta.SetEntityName(lpbravoUids[0], "Listening Point Bravo", meta);
-            _shuttle.SetIFFColor(lpbravoUids[0], lpbravoColor);
-            _shuttle.AddIFFFlag(lpbravoUids[0], IFFFlags.HideLabel);
-        }
 
-        // if (_map.TryLoad(mapId, northpole, out var northpoleUids, new MapLoadOptions
-        //     {
-        //         Offset = _random.NextVector2(2150f, 3900f)
-        //     }))
-        // {
-        //     var meta = EnsureComp<MetaDataComponent>(northpoleUids[0]);
-        //     _shuttle.SetIFFColor(northpoleUids[0], lpbravoColor);
-        //     _shuttle.AddIFFFlag(northpoleUids[0], IFFFlags.HideLabel);
-        // }
-
-        if (_map.TryLoad(mapId, arena, out var depotUid5s, new MapLoadOptions
-            {
-                Offset = _random.NextVector2(2200f, 4200f)
-            }))
-        {
-            var meta = EnsureComp<MetaDataComponent>(depotUid5s[0]);
-            _meta.SetEntityName(depotUid5s[0], "The Pit", meta);
-            _shuttle.SetIFFColor(depotUid5s[0], civilianColor);
-        }
+        var mapId = GameTicker.DefaultMap;
+        var dungenTypes = _prototypeManager.EnumeratePrototypes<DungeonConfigPrototype>();
 
-        if (_map.TryLoad(mapId, cove, out var depotUid6s, new MapLoadOptions
-            {
-                Offset = _random.NextVector2(10000f, 15000f)
-            }))
+        foreach (var dunGen in dungenTypes)
         {
-            if (_prototypeManager.TryIndex<GameMapPrototype>("Cove", out var stationProto))
+
+            var seed = _random.Next();
+            var offset = _random.NextVector2(3000f, 8500f);
```

Comment:
Shouldn't the dungeon respect the map scale?
```suggestion
            var offset = _random.NextVector2(3000f, 8500f) * _distanceOffset;
```

Reply:
yes, i just hadnt gotten to the changes for dungeons yet